frontend/packages/kubevirt-plugin/src/k8s/patches/vm/vm-patches.ts

```diff
@@ -34,3 +186,52 @@ export const getUpdateDescriptionPatches = (
   }
   return patches;
 };
+
+export const getUpdateFlavorPatches = (
+  vmLike: VMLikeEntityKind,
+  templates: TemplateKind[],
+  flavor: string,
+  cpu?: number,
+  mem?: string,
+): Patch[] => {
+  const patches = [];
+
+  // TODO: vm.kubevirt.io/template label should be changed as well (vm.kubevirt.io/template: win2k12r2-server-large)
+  // TODO: by changing flavor and so the base template, VM devices can be changed as well and so "full delta" should be applied here.
+  // Considering recent kubevirt state, updating cpu and sockets is good enough for now, but should be enhanced in the future.
+
+  let template;
+  if (isVM(vmLike)) {
+    template = getTemplateForFlavor(templates, vmLike as VMKind, flavor);
+  } else {
+    const vm = selectVM(vmLike as TemplateKind);
+    template = getTemplateForFlavor(templates, vm, flavor);
+  }
+
+  // flavor is set on object.metadata.label level for both VM and VMTemplate
+  patches.push(...getUpdateFlavorPatch(vmLike, flavor));
+
+  let customCpu = {
+    sockets: 1,
+    cores: cpu,
+    threads: 1,
+  };
+  let customMem = mem;
+  if (flavor !== CUSTOM_FLAVOR) {
+    const templateVm = selectVM(template);
+    customCpu = getCPU(templateVm) as CPU;
```

Comment:
this is acually CPURaw. The type should be added directly to selector.

Reply:
done